src/components/application_manager/test/commands/mobile/get_system_capability_request_test.cc

```diff
@@ -0,0 +1,246 @@
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  (*msg)[am::strings::msg_params][am::strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::REMOTE_CONTROL;
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+  EXPECT_CALL(app_mngr_, hmi_capabilities()).WillOnce(ReturnRef(mock_hmi_cap));
+  EXPECT_CALL(mock_hmi_cap, rc_capability()).WillRepeatedly(Return(&sm));
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS),
+                                  _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest,
+       Run_SystemCapabilityVideoStreaming_SUCCESS) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  (*msg)[am::strings::msg_params][am::strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::VIDEO_STREAMING;
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+  EXPECT_CALL(app_mngr_, hmi_capabilities()).WillOnce(ReturnRef(mock_hmi_cap));
+  EXPECT_CALL(mock_hmi_cap, video_streaming_capability())
+      .WillRepeatedly(Return(&sm));
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS),
+                                  _)).WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest, Run_DefaultCase_UnsupportedResource) {
+  MockAppPtr mock_app = CreateMockApp();
+  SmartObject sm(smart_objects::SmartType_Map);
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  (*msg)[am::strings::msg_params][am::strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::INVALID_ENUM;
+  SharedPtr<GetSystemCapabilityRequest> command(
+      CreateCommand<GetSystemCapabilityRequest>(msg));
+  EXPECT_CALL(app_mngr_, hmi_capabilities()).WillOnce(ReturnRef(mock_hmi_cap));
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResultCodeIs(mobile_result::UNSUPPORTED_RESOURCE), _))
+      .WillOnce(Return(true));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(GetSystemCapabilityRequestTest,
+       Run_HmiNotReturnSmartObject_DataNotAvailable) {
+  MockAppPtr mock_app = CreateMockApp();
+  MessageSharedPtr msg = CreateMessageParams();
+  amt::MockHMICapabilities mock_hmi_cap;
+  uint32_t size = sizeof(mobile_apis::SystemCapabilityType::eType);
```

Comment:
@ValeriiMalkov Const ? What is the sense to get size of type?

Reply:
@AByzhynar fixed at the commit ab40e21d3a66941f9a9d9d0a0eac8e18f09add44